exercises/practice/bob/.approaches/answer-array/content.md

```diff
@@ -0,0 +1,108 @@
+which is the size of a pointer for that platform.
+For a 64-bit system, the size of a pointer is 64 bits.
+So, each `str` is defined as a `str` reference (`&str`).
+The array itself is defined as a reference, since the compiler does not know the size of the whole array and its contents.
+
+- The [`str`][str] method [`trim_end`][trim-end] is used to remove whitespace from the end of the input.
+- If the trimmed input [`is_empty`][is-empty], then the response for nothing said is returned.
+- The [`ends_with`][ends-with] method is used to determine if the input ends with a question mark.
+If so, `is_questioning` is given the value `1`, otherwise it is given the value `0`.
+
+- The first half of the yell condition
+
+```rust
+message.chars().any(|ch| ch.is_alphabetic())
+```
+
+calls the `str` method [`chars`][chars] to create an [`Iterator`][iterator] for the characters in the trimmed input.
+- The iterated characters are passed into the `Iterator` method [`any`][any].
+- `any` passes each character to a [closure][closure] using the [`char`][char] method [`is_alphabetic`][is-alphabetic] to ensure there is at least one letter character in the `str`.
+This is because the second half of the condition tests that the uppercased input is the same as the input.
+If the input were only `"123"` it would equal itself uppercased, but without letters it would not be a yell.
+The uppercasing is done by using the `str` method [`to_uppercase`][to-uppercase].
+If the input is a yell, then `is_yelling` is given the value `2`, otherwise it is given the value `0`.
+
+The final expression returns the value in the `ANSWERS` array at the index of the combined values for `is_questioning` and `is_yelling`.
+
+```exercism/note
+Note that the final line is just `ANSWERS[is_questioning + is_yelling]`
+This is because the [last expression can be returned](https://doc.rust-lang.org/rust-by-example/fn.html)
+from a function without using `return` and a semicolon.
+```
+
+| is_yelling | is_questioning | Index     | Answer                                |
+| ---------- | -------------- | --------- | ------------------------------------- |
+| `false`    | `false`        | 0 + 0 = 0 | `"Whatever."`                         |
+| `false`    | `true`         | 0 + 1 = 1 | `"Sure."`                             |
+| `true`     | `false`        | 2 + 0 = 2 | `"Whoa, chill out!"`                  |
+| `true`     | `true`         | 2 + 1 = 3 | `"Calm down, I know what I'm doing!"` |
+
+## Shortening
+
+For defining the array the `static` lifetime specifiers can be dropped, like so
+
+```rust
+const ANSWERS: &[&str] = &[
+    "Whatever.",
+    "Sure.",
+    "Whoa, chill out!",
+    "Calm down, I know what I'm doing!",
+];
+```
+
+A lifetime specifier doesn't change the actual [lifetime][lifetime] of a binding.
+It just makes the lifetime explicit.
+When the compiler can figure out the lifetime, then the annotation can be dropped, which is called [lifetime elision][lifetime-elision].
```

Comment:
It's tempting, but since it comprises practically the entire text of the Shortening section. I'm thinking to leave it as is, at least for now. I might change my mind later on.